components/sup/src/manager/mod.rs

```diff
@@ -592,30 +616,24 @@ impl Manager {
                 .write()
                 .expect("Services lock is poisoned!")
                 .iter_mut() {
-            if self.updater
-                   .check_for_updated_package(service, &self.census_ring) {
-                service.populate(&self.census_ring);
-                self.gossip_latest_service_rumor(&service);
```

Comment:
Yep! This should be completely unnecessary. We are going to call `tick()` pretty much right after this on the service which is going to tell us that it updated. Once it does, we will call `gossip_latest_service_rumor()` as a very next step